### PR TITLE
ACM-10318 Upgrade risk predictions console backend

### PR DESCRIPTION
**Pull request by zlayne**

### CODE REVIEW

File: backend/src/routes/events.ts [hunk 1]
```diff
@@ -119,6 +119,11 @@ const definitions: IWatchOptions[] = [
   // **Need to look for creds with: 'cluster.open-cluster-management.io/type': 'ans', for edit scenarios
   { kind: 'Secret', apiVersion: 'v1', labelSelector: { 'cluster.open-cluster-management.io/type': 'ans' } },
   { kind: 'Secret', apiVersion: 'v1', fieldSelector: { 'metadata.name': 'auto-import-secret' } },
+  {
+    kind: 'Secret',
+    apiVersion: 'v1',
+    fieldSelector: { 'metadata.name': 'pull-secret', 'metadata.namespace': 'openshift-config' },
+  },
```
Comment (KevinFCormier) [inline backend/src/routes/events.ts:126]: Is this unused?

Comment (zlayne) [reply backend/src/routes/events.ts:126]: Yes, this was leftover I will remove.

File: backend/src/routes/upgrade-risks-prediction.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        chucks.push(chuck)
+      })
+      req.on('end', async () => {
+        data = chucks.join()
+        const body = JSON.parse(data) as UpgradeRiskBody
+
+        // acm-operator version in User-Agent header doesn't matter - CCX only uses the 'acm-operator' string to identify the product initiating the req
+        // https://github.com/RedHatInsights/insights-results-smart-proxy/blob/master/server/router_utils.go#L168
+        const userAgent = 'acm-operator/v2.10.0 cluster/acm-hub'
+        const insightsPath = 'https://console.redhat.com/api/insights-results-aggregator/v2/upgrade-risks-prediction'
```
[the remark this answers is not in the document]
Comment (zlayne) [reply backend/src/routes/upgrade-risks-prediction.ts:61]: Happy to add the necessary threat model information.
I'm not sure what the best strategy is on the issue of proxy environments. I can add an error for this case, but don't want to introduce a package that will require constant updates.

[the remark this answers is not in the document]
Comment (zlayne) [reply backend/src/routes/upgrade-risks-prediction.ts:62]: @KevinFCormier Added a change to use the `https-proxy-agent` package if `process.env.HTTPS_PROXY` is defined. Let me know what you think about this change.

[the remark this answers is not in the document]
Comment (KevinFCormier) [reply backend/src/routes/upgrade-risks-prediction.ts:62]: Sounds good. Thanks for addressing.

**Pull request merged**